spec/system/admin/reports_spec.rb

```diff
@@ -114,9 +105,9 @@
         ReportJob.perform_now(**args)
         breakpoint.synchronize { "continue after unlocked" }
       end
-      click_button "Go"
 
-      expect(page).to have_selector "#report-table table"
+      click_on "Go"
+
       expect(page).to have_content "FIRST NAME LAST NAME BILLING ADDRESS EMAIL"
```

Comment:
The mock above is executing the job inline with `perform_now` already. So there's no need to trigger job processing. What hasn't happened yet in this instance is that the controller action hasn't finished yet. The report has been displayed independently via cable ready while the controller is still stuck on the mutex lock.